yugabyte/src/yugabyte/ycql/bank_improved.clj

```diff
@@ -0,0 +1,92 @@
+(ns yugabyte.ycql.bank-improved
+  (:refer-clojure :exclude [test])
+  (:require [clojure.tools.logging :refer [debug info warn]]
+            [clojurewerkz.cassaforte.client :as cassandra]
+            [clojurewerkz.cassaforte.cql :as cql]
+            [clojurewerkz.cassaforte.query :as q :refer :all]
+            [yugabyte.ycql.client :as c]))
+
+(def setup-lock (Object.))
+(def keyspace "jepsen")
+(def table-name "accounts")
+(def counter-start (atom 0))
+(def counter-end (atom 0))
+
+(c/defclient CQLBank keyspace []
+  (setup! [this test]
+          (c/create-transactional-table
+           conn table-name
+           (q/if-not-exists)
+           (q/column-definitions
+            {:id          :int
+             :balance     :bigint
+             :primary-key [:id]}))
+          (info "Creating accounts")
+          (c/with-retry
+           (cql/insert-with-ks conn keyspace table-name
+                               {:id      (first (:accounts test))
+                                :balance (:total-amount test)})
+           (doseq [a (rest (:accounts test))]
+             (do
+               (swap! counter-end inc)
+               (cql/insert conn table-name
+                           {:id a, :balance 0})))))
+
+  (invoke! [this test op]
+           (c/with-errors op #{:read}
+                          (case (:f op)
+                            :read
+                            (->> (cql/select-with-ks conn keyspace table-name)
+                                 (map (juxt :id :balance))
+                                 (into (sorted-map))
+                                 (assoc op :type :ok, :value))
+
+                            :transfer
+                            (let [{:keys [from to amount]} (:value op)
+                                  ; dice                     (rand-nth ["insert" "update" "delete"])]
+                                  dice                     (rand-nth ["insert" "update"])]
+                              (cond
+                                (= dice "insert")
```

Comment:
This looks really weird, if not outright wrong.
Why does the client randomly choose what operations to perform? This should be part of the workload instead